vms/platformvm/block/builder/builder.go

```diff
@@ -103,24 +102,14 @@ func New(
 
 // AddUnverifiedTx verifies a transaction and attempts to add it to the mempool
 func (b *builder) AddUnverifiedTx(tx *txs.Tx) error {
-	if !b.txExecutorBackend.Bootstrapped.Get() {
-		return ErrChainNotSynced
-	}
-
 	txID := tx.ID()
 	if b.Mempool.Has(txID) {
```

Comment:
honestly also this mempool.Has check could be in MempoolTxVerifier. We could evolve the MempoolTxVerifier to include the full mempool, so to be able to carry out all relevant checks (including conflicts on zero confimation utxo expenditure in P-chain mempool, if we ever want to enable it)

Reply:
I'd like to defer making MempoolTxVerifier more bloated than it needs to be- this check is being moved to `network.issueTx` in #2296. This PR is a stepping stone towards that